### PR TITLE
feat(ssg): Implement Dynamic File Extension on MIME Type and Enhanced Response Handling

### PR DESCRIPTION
**Pull request by watany-dev**
fixes: https://github.com/honojs/hono/issues/1962 https://github.com/honojs/hono/issues/1963

I have addressed these two issues together as they are interconnected

### Author should do the followings, if applicable

- [x] Add tests
- [x] Run tests
- [x] `yarn denoify` to generate files for Deno

### CODE REVIEW

**Comment by watany-dev** [comment]
If it's necessary to implement all corresponding MIME types, the determineExtension function might become too large, so I'm considering moving it to utils.ts.

File: deno_dist/helper/ssg/index.ts [hunk 1]
```diff
@@ -25,57 +25,93 @@ export interface ToSSGResult {
...
   return joinPaths(outDir, fileName)
 }
 
+const parseResponseContent = async (response: Response): Promise<string | ArrayBuffer> => {
+  const contentType = response.headers.get('Content-Type')
+
+  try {
+    if (contentType?.includes('text')) {
+      return await response.text()
+    } else if (contentType?.includes('json')) {
```
Comment (yusukebe) [inline deno_dist/helper/ssg/index.ts:40]: Maybe we can use `response.text()` for json also? I don't think it is necessary to decode JSON and then turn it into a string.

File: deno_dist/helper/ssg/index.ts [hunk 2]
```diff
@@ -25,57 +25,93 @@ export interface ToSSGResult {
...
+      return await response.arrayBuffer()
+    }
+  } catch (error) {
+    throw new Error(
+      `Error processing response: ${error instanceof Error ? error.message : 'Unknown error'}`
+    )
+  }
+}
+
+const determineExtension = (mimeType: string): string => {
```
Comment (yusukebe) [inline deno_dist/helper/ssg/index.ts:52]: How about using the `utils/mime.ts`?

```diff
diff --git a/src/helper/ssg/index.ts b/src/helper/ssg/index.ts
index 125f2cd..c7f4dc2 100644
--- a/src/helper/ssg/index.ts
+++ b/src/helper/ssg/index.ts
@@ -1,6 +1,7 @@
 import { inspectRoutes } from '../../helper/dev'
 import type { Hono } from '../../hono'
 import type { Env, Schema } from '../../types'
+import { getExtension } from '../../utils/mime'
 import { joinPaths, dirname } from './utils'

 /**
@@ -49,6 +50,8 @@ const parseResponseContent = async (response: Response): Promise<string | ArrayB
 }

 const determineExtension = (mimeType: string): string => {
+  const extension = getExtension(mimeType)
+  if (extension) return extension
   switch (mimeType) {
     case 'text/html':
       return '.html'
diff --git a/src/utils/mime.ts b/src/utils/mime.ts
index 5b55663..807cec9 100644
--- a/src/utils/mime.ts
+++ b/src/utils/mime.ts
@@ -9,6 +9,12 @@ export const getMimeType = (filename: string): string | undefined => {
   return mimeType
 }

+export const getExtension = (mimeType: string): string | undefined => {
+  const extensions = Object.keys(mimes)
+  const extension = extensions.find((ext) => mimes[ext] === mimeType)
+  return extension
+}
+
 const mimes: Record<string, string> = {
   aac: 'audio/aac',
   abw: 'application/x-abiword',
```

**Comment by yusukebe** [comment]
Hi @watany-dev 

Almost seems to be good, but I've left the comments. Check them.

**Comment by watany-dev** [comment]
@yusukebe 
Thank you for the review. I forgot about `util/mime`.

**Review by yusukebe — APPROVED** [review]
Looks great to me!

**Comment by yusukebe** [comment]
@watany-dev 

Thanks very much. Merging now.

**Pull request merged**